clang/lib/Sema/SemaOverload.cpp

```diff
@@ -1511,8 +1511,11 @@ static bool IsOverloadOrOverrideImpl(Sema &SemaRef, FunctionDecl *New,
       auto NewObjectType = New->getFunctionObjectParameterReferenceType();
       auto OldObjectType = Old->getFunctionObjectParameterReferenceType();
 
-      if (NewObjectType.isConstQualified() != OldObjectType.isConstQualified())
-        return false;
+      if (Old->isExplicitObjectMemberFunction() &&
+          OldObjectType.getQualifiers() != NewObjectType.getQualifiers())
```

Comment:
do we care about `__restrict` @AaronBallman ?

Reply:
The always-early-return doesn't make sense here to me, as Corentin mentioned, because `__restrict` can't be overloaded on IIRC.